x/authority/keeper/keeper.go

```diff
@@ -42,31 +44,45 @@ func NewKeeper(cdc codec.BinaryMarshaler, storeKey sdk.StoreKey, issuerKeeper is
 	}
 }
 
-func (k Keeper) SetAuthority(ctx sdk.Context, authority sdk.AccAddress) {
-	store := ctx.KVStore(k.storeKey)
-
-	if store.Get([]byte(keyAuthorityAccAddress)) != nil {
-		panic("Authority was already specified")
+func (k Keeper) SetAuthority(ctx sdk.Context, newAuthority sdk.AccAddress) {
```

Comment:
Replace the genesis call with a default config encrypted key scheme? i.e. IBC style. Once an authority is set then disable the file store key.

Reply:
To summarize my understanding of what we agreed on during the call:

This function should allow only one call and then be sealed for further calls. Consider whether to `panic` in case of further invocations. The goal is that the function should only be invoked during initialization from genesis.

This would require `replaceAuthority` to be modified to no longer use `SetAuthority`.

Reply:
@haasted  this change does not accommodate against immediate disassociation of former authorities. After you replace the authority for up to the renewed 24-hour transition window, the former authority may inconspicuously set a new stealth authority!